sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -318,9 +318,20 @@ specificFunction
     ;
 
 relevanceFunction
-    : relevanceFunctionName LR_BRACKET
-        field=relevanceArgValue COMMA query=relevanceArgValue
-        (COMMA relevanceArg)* RR_BRACKET
+    : singleFieldRelevanceFunction | multiFieldRelevanceFunction
+    ;
+
+// Field is a single column
+singleFieldRelevanceFunction
+    : singleFieldRelevanceFunctionName LR_BRACKET
+        field=relevanceField COMMA query=relevanceQuery
+        (COMMA relevanceArg)* RR_BRACKET;
+
+// Field is a list of columns
+multiFieldRelevanceFunction
+    : multiFieldRelevanceFunctionName LR_BRACKET
+        LT_SQR_PRTHS field=relevanceFieldAndWeight (COMMA field=relevanceFieldAndWeight)* RT_SQR_PRTHS
```

Comment:
I believe if we add a separate rule for field list (everything between square brackets), we can simplify AstExpressionBuilder.